### PR TITLE
Added adr doc for supporting multi env repos

### PR DESCRIPTION
**Pull request by HumairAK**
The following PR's are in accordance with this new proposal: 
- https://github.com/operate-first/argocd-apps/pull/14
- https://github.com/operate-first/odh/pull/46

### CODE REVIEW

File: docs/adr/0005-support-multi-environments-in-repos.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Repositories supposrting multiple environment deployments
```
Comment (anishasthana) [inline docs/adr/0005-support-multi-environments-in-repos.md]: `supposrting`

**Review by anishasthana — CHANGES_REQUESTED** [review]
Requested a small change. As a whole I think this really makes sense

File: docs/adr/0005-support-multi-environments-in-repos.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+1) Keep argocd-apps repo cluster specific, and use multiple repos as "app-of-apps" repos for each cluster/environment
+2) Use Kustomize overlays/bases for argocd-apps repo and have different overlays that patch the `path` for each cluster/environment
+
+## Decision Outcome
+Choose 2) as it accomplishes much of 1) without needing yet more repos and adding bloat to the organization. 
+
+The `argocd-apps` repo would follow the following structure: 
+
+The example makes these assumptions: 
+- There are 2 target clusters: `cluster_1` and `cluster_2`
```
Comment (goern) [inline docs/adr/0005-support-multi-environments-in-repos.md]: I'm a little confused by "cluster", isnt it simply an "environment", that could be two environments running on the same OpenShift cluster?!

Comment (HumairAK) [reply docs/adr/0005-support-multi-environments-in-repos.md]: Yeah it can, I'll change cluster to environments.

Comment (HumairAK) [reply docs/adr/0005-support-multi-environments-in-repos.md]: updated to be more general: changed cluster -> environments

**Review by tumido — APPROVED** [review]
LGTM :+1:

**Review by anishasthana — APPROVED** [review]

**Comment by HumairAK** [comment]
If we're happy with this can we get this merged? 

@durandom  -- wdyt?

**Review by durandom — APPROVED** [review]
Very well written ❤️

**Pull request merged**